User can create a tweet

foxed opened the pull request:
Sorry, I wasn't really good committing my work in a manner that made sense (like, "add controller" "add tests" etc). But there's not much code so hope that's alright. And I will squash those two indentation commits before merging.

@dorothythurston

dorothythurston commented on db/migrate/20180608204308_create_tweets.rb:
```diff
@@ -0,0 +1,10 @@
+class CreateTweets < ActiveRecord::Migration[5.2]
+  def change
+    create_table :tweets do |t|
+      t.string :tweet
```
should this be `:user` ?

also we should add a foreign key check on this to make sure that the user id exists and an index on user id... i think the syntax is something like `add_reference :tweets, :user, foreign_key: true` which will add the foreign key and the index i think but that might be outdated

dorothythurston replied:
oh i see you did the adding user to the migration in a separate migration. word.

dorothythurston replied:
wait so what is the difference between the tweet and `string` columns? i feel like maybe we only need one of the two columns?

foxed replied:
no the string column was an accident i just noticed, i prob typed it incorrectly when i generated teh migration i'll fix that

dorothythurston replied:
i think you can go ahead and just modify your original migration because it isn’t merged.
so i’d say remove the db/migrate/20180612211841_remove_string_from_tweet.rb and then modify db/migrate/20180608204308_create_tweets.rb
also maybe we should call it like tweet.body instead of tweet.tweet
just thinking of names names names
other than that this is looking tight

dorothythurston commented on spec/controllers/tweets_controller_spec.rb:
```diff
@@ -0,0 +1,10 @@
+
+RSpec.describe TweetsController, :type => :controller do
+  it "assigns tweets" do
+  	user = create :user
```
looks like this got indented one tab too far here

dorothythurston commented on spec/controllers/tweets_controller_spec.rb:
```diff
@@ -0,0 +1,10 @@
+require "rails_helper"
+
+
+RSpec.describe TweetsController, :type => :controller do
```
i think we can update this hash syntax to `type: :controller`

dorothythurston commented:
this is looking very good! just a couple of clarification questions but other than that solid work

dorothythurston commented on spec/controllers/tweets_controller_spec.rb:
```diff
@@ -0,0 +1,10 @@
+require "rails_helper"
+
+
```
extra line space here

dorothythurston commented on spec/factories/factories.rb:
```diff
@@ -0,0 +1,6 @@
+FactoryBot.define do
+  factory :tweet do
+    tweet "hello i am a tweet"
```
so i think this should be body now right?

dorothythurston commented on db/migrate/20180614152342_change_column_name.rb:
```diff
@@ -0,0 +1,5 @@
+class ChangeColumnName < ActiveRecord::Migration[5.2]
+  def change
+  	rename_column :tweets, :tweet, :body
```
i think since its not merged you can go ahead and modify the original migration

dorothythurston commented on db/schema.rb:
```diff
@@ -30,6 +30,14 @@
+    t.string "body"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.bigint "user_id"
```
Do we want tweets to be able to be created if they do not have a user? Should this have a null: false as well?

foxed replied:
yes, i also added null: false on the body, which (if i remember correctly) just means that a tweet can't be blank...right?

dorothythurston commented on db/schema.rb:
```diff
@@ -30,6 +30,14 @@
   end
 
+  create_table "tweets", force: :cascade do |t|
+    t.string "body"
```
Do we want to throw a length limit on this at a DB level?

dorothythurston committed:
tweet belongs_to user user has_many tweets

tweet must have user, tweet cannot be blank, tweet must be 280 chars or less

validates presence of body on tweet

dorothythurston commented:
Nice. Rebase this all up into one commit?

dorothythurston commented:
@foxed yeah looks like there are some conflicts here that i introduced and github can't resolve easily

foxed commented:
@dorothythurston good to merge?

dorothythurston reviewed (approved):
This is looking good, You could add a validation of presence and limit for the tweet body so it hits that first before going to the DB  but other than that 👍

foxed commented:
@dorothythurston is this what you're looking for ? (refer to the updated tweet model). we can also add an error message if the length is too long, but i wasn't sure if it would be better doing that there or a dynamic character counting system we can add later on the view (like how twitter does it). but lmk i can change it

dorothythurston commented:
@foxed yeah exactly!

dorothythurston commented:
i approved this so whenever you are ready feel free to rebase and merge

foxed commented:
@dorothythurston done, sorry for the delay
